### PR TITLE
Adding getViewData method call

### PR DESCRIPTION
**Pull request by ecwillis**
Adding a conditional call to getViewData to set context data per request.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -181,7 +181,10 @@ Rapptor.prototype._setupViews = function() {
     if (response.variety === 'view') {
 
       var context = response.source.context || {};
-      _.defaults(context, self._viewData);
+      if (typeof this.getViewData === 'function') {
+        this.getViewData(request, context);
+      }
+      _.defaults(context, this._viewData);
```
Comment (jgallen23) [inline index.js]: lets kill this._viewData and just set the defaults above line 184

Comment (jgallen23) [inline index.js]: just basically context.env = config.env

Comment (jgallen23) [inline index.js]: then remove the setViewData function and the _viewData object

Comment (ecwillis) [inline index.js]: How would that affect the fbID from here?

```
rapptor.setViewData(function(config) {
  return {
    facebookAppId: config.facebook.appId,
    env: config.env
  };
});
```

**Comment by jgallen23** [comment]
we should add config in there too:

usage:
```
rapptor.getViewData(function(request, context, config) {
  context.facebookAppId = config.facebook.appId,
});
```

**Pull request merged**